File: src/Vincent OS/Game.vb
﻿Public Class game

    Private Sub game_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Bureau.Visible = True Then
            Bureau.Button22.Visible = True
        ElseIf Bureau2.Visible = True Then
            Bureau2.Button22.Visible = True
        End If
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        exewin.Show()
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        MazeGame.Show()
    End Sub

    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        ''Si aucun jeu n'est détecté, commenter
        ''Store.Show()
        Dim errorsound As New System.Media.SoundPlayer(My.Resources.errorSound)
        errorsound.Play()
        MessageBox.Show("ERREUR : Aucun jeu supplémentaire n'as été détecté sur le Store.", "ERREUR", MessageBoxButtons.OK, MessageBoxIcon.Error)
    End Sub

    Private Sub game_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed
        If Bureau.Visible = True Then
            Bureau.Button22.Visible = False
        ElseIf Bureau2.Visible = True Then
            Bureau2.Button22.Visible = False
        End If
    End Sub
End Class

File: src/Vincent OS/Game.cs
using System;
using System.Windows.Forms;

namespace Vincent_OS
{
    public partial class game
    {
        public game()
        {
            InitializeComponent();
        }

        private void game_Load(object sender, EventArgs e)
        {
            Bureau bureau = new Bureau();
            Bureau2 bureau2 = new Bureau2();
            if (bureau.Visible == true)
            {
                bureau.Button22.Visible = true;
            }
            else if (bureau2.Visible == true)
            {
                bureau2.Button22.Visible = true;
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            exewin exewin = new exewin();
            exewin.Show();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            MazeGame mazeGame = new MazeGame();
            mazeGame.Show();
        }

        private void game_FormClosed(object sender, FormClosedEventArgs e)
        {
            Bureau bureau = new Bureau();
            Bureau2 bureau2 = new Bureau2();
            if (bureau.Visible == true)
            {
                bureau.Button22.Visible = false;
            }
            else if (bureau2.Visible == true)
            {
                bureau2.Button22.Visible = false;
            }
            GC.Collect();
            GC.WaitForPendingFinalizers();
        }
    }
}